src/bin/ion/commands/dump.rs

```diff
@@ -35,9 +37,12 @@ pub fn app() -> Command {
 pub fn run(_command_name: &str, matches: &ArgMatches) -> Result<()> {
     // --format pretty|text|lines|binary
     // `clap` validates the specified format and provides a default otherwise.
-    let format = matches
-        .get_one::<String>("format")
-        .expect("`format` did not have a value");
+    let format = matches.get_one::<String>("format").unwrap();
+
+    // --values <n>
+    // this value is supplied when `dump` is invoked as `head`
+    let values = *matches.try_get_one::<i32>("values")
```

Comment:
(Suggestion) We should probably shift `head` related logic to `head.rs`. Whatever part of `dump` we are reusing there we should create a helper method for that in `dump.rs` and then use it in `head.rs`.

Example:
```rust
// dump.rs
pub fn run() {
    ...
    write_ion_values(...) // a helper method that can also be used in head.rs
    ...
}

pub fn run() {
   ...
   // perform the values option(--values <n>) extraction here and use expect if the option is not found
   write_ion_values(...)
   ...
}
```

Reply:
Backlogged as:
* https://github.com/amazon-ion/ion-cli/issues/49